src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1190,12 +1315,32 @@ else if (geometry instanceof LineString || geometry instanceof Polygon)
         }
     }
 
+    private List<CountryBoundary> boundariesHelper(final Supplier<List<Polygon>> supplier,
```

Comment:
this seems to be a critical function for many of the methods above. Could we get a comment for this function?